ugs-core/src/com/willwinder/universalgcodesender/listeners/ControllerStatusBuilder.java

```diff
@@ -107,12 +134,84 @@ public ControllerStatusBuilder setPins(ControllerStatus.EnabledPins pins) {
         return this;
     }
 
-    public ControllerStatusBuilder setStates(ControllerStatus.AccessoryStates states) {
-        this.states = states;
+    public ControllerStatusBuilder setAccessoryStates(ControllerStatus.AccessoryStates states) {
+        this.accessoryStates = states;
         return this;
     }
 
+    public static ControllerState getControllerStateFromStateString(String stateString) {
```

Comment:
The method for creating a controller state from string is hardware dependant and should not exist in the generic ControllerStatus.

I was hoping that the state string would be removed in favour of the more static enum types.

Reply:
I see, let me rework this :)